.github/actions/run-ios-simulator/src/main.ts

```diff
@@ -0,0 +1,74 @@
+    const id = uuidv4().split("-").join("");
+
+    try {
+        const appPath = core.getInput("appPath", { required: true });
+        const bundleId = core.getInput("bundleId", { required: true });
+        const iphoneToSimulate = core.getInput("iphoneToSimulate", { required: false });
+        const args = core.getInput("arguments", { required: false });
+
+        let runtimeId = "";
+        const options : exec.ExecOptions = {};
+        options.listeners = {
+            stdout: (data: Buffer) => {
+                runtimeId += data.toString();
+            },
+        };
+
+        await execCmd("xcrun simctl list runtimes", options);
+
+        // Sample output: iOS 14.5 (14.5 - 18E182) - com.apple.CoreSimulator.SimRuntime.iOS-14-5
+        // and we want to extract "iOS 14.5"
+        // If we want to allow launching watchOS/tvOS simulators, replace the 'iOS' with an 'os' argument
+        const matches = runtimeId.match(/(iOS \d{1,2}(.\d{1,2})?)/g);
+        if (matches && matches.length > 0) {
+            runtimeId = matches[0].replace(" ", "");
+            core.info(`runtimeId: ${runtimeId}`);
+        }
+
+        try {
+            await execCmd(`xcrun simctl create ${id} com.apple.CoreSimulator.SimDeviceType.${iphoneToSimulate} ${runtimeId.toString()}`);
+        }
+        catch {
+            // Different combinantions of xcode and macOS versions have shown different syntax acceptance about the runtime, therefore 1 last attempt with a different syntax.
+            // additionally, this uses  childProcess.exec instead of @actions/exec because the latter doesn't properly parse the pipe (|) char
+            const { stdout, stderr } = await childProcess.exec("xcrun simctl list runtimes |  awk '/com.apple.CoreSimulator.SimRuntime.iOS/ { match($0, /com.apple.CoreSimulator.SimRuntime.iOS-[0-9.-]+/); print substr($0, RSTART, RLENGTH); exit }'");
+            runtimeId = stdout?.toString() ?? "";
+            if (stderr) {
+                core.setFailed(stderr.toString());
+            }
+
+            await execCmd(`xcrun simctl create ${id} com.apple.CoreSimulator.SimDeviceType.${iphoneToSimulate} ${runtimeId.toString()}`);
+        }
+
+        await execCmd(`xcrun simctl boot ${id}`);
+        await execCmd(`xcrun simctl install ${id} ${appPath}`);
+        await execCmd(`xcrun simctl launch --console-pty ${id} ${bundleId} ${args}`);
+    } catch (error) {
+        core.setFailed(error.message);
+    } finally {
+        try {
+            await execCmd(`xcrun simctl shutdown ${id}`);
+            await execCmd(`xcrun simctl delete ${id}`);
+        } catch (error) {
+            core.setFailed(`An error occurred during cleanup: ${error.toString()}`);
+        }
+    }
+}
+
+async function execCmd(cmd: string, options: exec.ExecOptions = {}): Promise<void> {
+    const exitCode = await exec.exec(cmd, [], options);
+    if (exitCode != 0) {
+        const msgCmd = cmd.split(" ").slice(0, 3).join(" ");
```

Comment:
Why are we removing some of the arguments here - while they'll probably be visible in the logs, it's still useful to have the complete command when an error occurs.

Reply:
Exactly because they are be visible in the logs. Every command executed, regardless of the result, is fully shown in the log. I thought to keep the err message concise since the full cmd is likely a few lines above. But if you want I'll fully print it out.

Reply:
I think it won't become overly verbose - let's print the entire thing so that the error is immediately visible in the status rather than having to go through the logs.